File: Java-CodeServer/src/main/resources/js/prompto/type/EnumeratedCategoryType.js
var CategoryType = require("./CategoryType").CategoryType;


function EnumeratedCategoryType(name) {
	CategoryType.call(this, name);
	return this;
}

EnumeratedCategoryType.prototype = Object.create(CategoryType.prototype);
EnumeratedCategoryType.prototype.constructor =  EnumeratedCategoryType;

/*
	@Override
	public IType checkMember(Context context, String name) throws SyntaxError {
		if ("value".equals(name))
			return this;
		else if ("name".equals(name))
			return TextType.instance();
		else
			return super.checkMember(context, name);
	}

*/

exports.EnumeratedCategoryType = EnumeratedCategoryType;

File: Java-Core/src/main/java/prompto/type/EnumeratedCategoryType.java
package prompto.type;

import prompto.error.SyntaxError;
import prompto.grammar.Identifier;
import prompto.runtime.Context;


public class EnumeratedCategoryType extends CategoryType {

	public EnumeratedCategoryType(Identifier id) {
		super(id); 
	}
	
	@Override
	public IType checkMember(Context context, Identifier id) {
		String name = id.toString();
		if ("value".equals(name))
			return this;
		else if ("name".equals(name))
			return TextType.instance();
		else
			return super.checkMember(context, id);
	}

}
